### DISCUSSION TITLE
dbus rather than busctl

### DISCUSSION THREAD

**Discussion by ghost** [rauc/rauc#1234] [General]
I have seen the documentation regarding busctl and in particular the following commands:

Obtain bundle information
```
busctl call de.pengutronix.rauc / de.pengutronix.rauc.Installer InspectBundle sa{sv} "server url" 0
```

Triggering an installation
```
busctl call de.pengutronix.rauc / de.pengutronix.rauc.Installer InstallBundle sa{sv} "server url" 0
```

Get the Operation property containing the current operation
```
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Operation
```

Get the Progress property containing the progress information
```
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Progress
```

Get the LastError property, which contains the last error that occurred during an installation
```
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer LastError
```

From the above how can exercise the above commands using dbus and in particular i have a c application that I want to make either busctl or dbus calls; how do I achieve this?

**Comment by ejoerns** [comment]
That is a question that is probably out-of-scope for support here because it depends on basic knowledge about Linux systems as well as project-specific decisions on libraries to use and the specific usage of these libraries.

The above commands actually use D-Bus, that's what they are made for. There exists a number of libraries that offer D-Bus bindings. You can find an overview here:

https://www.freedesktop.org/wiki/Software/DBusBindings/

For RAUC, as you can see from the code, we use gdbus. Examples of how external clients can use the D-Bus API can be found e.g. in [contrib/cgi/](https://github.com/rauc/rauc/tree/master/contrib/cgi) folder or in [rauc-hawkbit-updater](https://github.com/rauc/rauc-hawkbit-updater/blob/master/src/rauc-installer.c).
